### DISCUSSION TITLE
question about DIO1 for sx127x

### DISCUSSION THREAD

**Discussion by 4rtama** [jgromes/RadioLib#1273] [Q&A] [answered]
I want to ask whether the SX127x if you want to change the mode to FSK must use the DIO1 pin ? because I have tried using only (NSS, DIO0, RST) only LORA can be used

**Comment by jgromes** [comment]
It depends on what you're trying to do. For basic sending and receiving packets, DIO0 is enough as it is used to signal both end of transmission as well as new packet received. For more advanced things like stream or direct mode, additional DIO pins are needed.

**Reply by 4rtama** [reply]
but I have tried it and the result is error -101 and -16
![Screenshot 2024-10-15 032225](https://github.com/user-attachments/assets/f348d713-7fe9-4b58-bb3a-7f718a79edd0)

**Reply by jgromes** [reply, marked as answer]
What leads you to believe either of these errors is related to DIO1 not being connected? You should probably try to find out what those errors actually mean (for example, in the documentation: https://jgromes.github.io/RadioLib/).

**Reply by 4rtama** [reply]
initially I tried in LoRa mode and it worked but when I wanted to change to FSK there was an error -16 because there are not many tutorials about it so I assume that DIO1 is important because when I first used radiolib there were 4 pins there (NSS, DIO0, RST and DIO1) (initially I used the Lora.h library mostly only using NSS, DIO0, RST) and what does this mean can be caused by a damaged SPI cable? are there any suggestions because I want to make it a research material in college

sorry if the grammar is bad I use google translate

**Reply by HeadBoffin** [reply]
You were being directed to find out what those error numbers mean - which you can do by **searching the library**.

You will also find more people helping if you post text as text, not as an image ...
